moveit_planners/ompl/ompl_interface/test/test_state_validity_checker.cpp

```diff
@@ -0,0 +1,361 @@
+
+    // set the default weight to avoid warning in test output
+    position_constraint.weight = 1.0;
+
+    return position_constraint;
+  }
+
+  ompl_interface::ModelBasedStateSpacePtr state_space_;
+  ompl_interface::ModelBasedPlanningContextSpecification planning_context_spec_;
+  ompl_interface::ModelBasedPlanningContextPtr planning_context_;
+  planning_scene::PlanningScenePtr planning_scene_;
+};
+
+// /***************************************************************************
+//  * Run all tests on the Panda robot
+//  * ************************************************************************/
+class PandaValidity : public TestStateValidityChecker
+{
+protected:
+  PandaValidity() : TestStateValidityChecker("panda", "panda_arm")
+  {
+  }
+};
+
+TEST_F(PandaValidity, testConstructor)
+{
+  testConstructor();
+}
+
+TEST_F(PandaValidity, testJointLimits)
+{
+  // use the panda "ready" state from the srdf config
+  // we know this state should be within limits and self-collision free
+  testJointLimits({ 0, -0.785, 0, -2.356, 0, 1.571, 0.785 });
+}
+
+TEST_F(PandaValidity, testSelfCollision)
+{
+  // the given state has self collision between "hand" and "panda_link2"
+  // (I just tried a couple of random states until I found one that collided.)
+  testSelfCollision({ 2.31827, -0.169668, 2.5225, -2.98568, -0.36355, 0.808339, 0.0843406 });
+}
+
+TEST_F(PandaValidity, testPathConstraints)
+{
+  // use the panda "ready" state from the srdf config
+  // we know this state should be within limits and self-collision free
+  testPathConstraints({ 0, -0.785, 0, -2.356, 0, 1.571, 0.785 });
+}
+
+/***************************************************************************
+ * Run all tests on the Fanuc robot
+ * ************************************************************************/
+class FanucTest : public TestStateValidityChecker
+{
+protected:
+  FanucTest() : TestStateValidityChecker("fanuc", "manipulator")
+  {
+  }
+};
+
+TEST_F(FanucTest, createStateValidityChecker)
+{
+  testConstructor();
+}
+
+TEST_F(FanucTest, testJointLimits)
+{
+  // I assume the Fanucs's zero state is within limits and self-collision free
+  testJointLimits({ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 });
+}
+
+TEST_F(FanucTest, testSelfCollision)
+{
+  // the given state has self collision between "base_link" and "link_5"
+  // (I just tried a couple of random states until I found one that collided.)
+  testSelfCollision({ -2.95993, -0.682185, -2.43873, -0.939784, 3.0544, 0.882294 });
+}
+
+TEST_F(FanucTest, testPathConstraints)
+{
+  // I assume the Fanucs's zero state is within limits and self-collision free
+  testPathConstraints({ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0 });
+}
+
+/***************************************************************************
+ * MAIN
+ * ************************************************************************/
+int main(int argc, char** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
```

Comment:
[nit] newline at end of file

Reply:
I think I added newlines now, but I can't see them on Github... Locally `git show` gives
```
 {
   testing::InitGoogleTest(&argc, argv);
   return RUN_ALL_TESTS();
-}
\ No newline at end of file
+}
```
and I see an extra line in my IDE.